libminifi/test/unit/StringUtilsTests.cpp

```diff
@@ -50,6 +50,16 @@ TEST_CASE("TestStringUtils::split4", "[test split classname]") {
   REQUIRE(expected == StringUtils::split(org::apache::nifi::minifi::core::getClassName<org::apache::nifi::minifi::utils::StringUtils>(), "::"));
 }
 
+TEST_CASE("TestStringUtils::split5", "[test split delimiter not specified]") {
```

Comment:
This used to be a SPoF. The current behaviour is now to keep the input string if the delimiter is `""`.

Reply:
There was a `delimiter.at(0)` check in the split implementation.

I will change it to split by characters. However, python and ruby split on whitespace by default:
https://www.w3schools.com/python/ref_string_split.asp
https://www.geeksforgeeks.org/ruby-string-split-method-with-examples/

Reply:
Default (no delimiter specified) and empty delimiter are different. Python fails on empty delimiter, ruby splits by characters.

The delimiter can have a default argument of `" "` (whitespace).

Reply:
I see, updated the function for splitting into characters on empty delimiter.

Reply:
Now this only tests with empty delimiter. Please add a test for when the delimiter is not specified.

Reply:
There is no default value for the delimiter. Updated the test description to say `"test split with delimiter set to empty string"`.